pkg/splunk/client.go

```diff
@@ -93,6 +92,10 @@ func newSplunkLogger(ctx context.Context, url, token, source, hostname string) *
 		},
 	}
 
+	if maximumSize != 0 {
+		sl.maximumSize = maximumSize
+	}
+
 	rcl.RetryWaitMin = 300 * time.Millisecond
 	rcl.RetryWaitMax = 3 * time.Second
 	rcl.RetryMax = 5
```

Comment:
I think we will ever only need to change size, maybe event size but I like your `strings.Repeat` trick so that will work. Channel lentgh should not be configurable, it does not matter too much.